ui/store/actions.ts

```diff
@@ -2457,60 +2456,88 @@ export function setProviderType(
+  },
+  {
+    setActive,
+    source,
+  }: {
+    setActive: boolean;
+    source?: string;
+  },
 ): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
-  return async (dispatch: MetaMaskReduxDispatch) => {
+  return async (dispatch) => {
     log.debug(
-      `background.updateAndSetCustomRpc: ${newRpcUrl} ${chainId} ${
-        ticker ?? EtherDenomination.ETH
-      } ${nickname}`,
+      `background.upsertNetworkConfiguration: ${rpcUrl} ${chainId} ${ticker} ${nickname}`,
     );
-
+    let networkConfigurationId;
     try {
-      await submitRequestToBackground('updateAndSetCustomRpc', [
-        newRpcUrl,
-        chainId,
-        ticker ?? EtherDenomination.ETH,
-        nickname || newRpcUrl,
-        rpcPrefs,
-      ]);
+      networkConfigurationId = await submitRequestToBackground(
+        'upsertNetworkConfiguration',
+        [
+          { rpcUrl, chainId, ticker, nickname: nickname || rpcUrl, rpcPrefs },
+          { setActive, source, referrer: ORIGIN_METAMASK },
+        ],
+      );
     } catch (error) {
-      logErrorWithMessage(error);
-      dispatch(displayWarning('Had a problem changing networks!'));
+      log.error(error);
+      dispatch(displayWarning('Had a problem adding network!'));
     }
+    return networkConfigurationId;
   };
 }
 
-export function editRpc(
-  oldRpcUrl: string,
-  newRpcUrl: string,
-  chainId: string,
-  ticker: EtherDenomination,
-  nickname: string,
-  rpcPrefs: RPCDefinition['rpcPrefs'],
-): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
-  return async (dispatch: MetaMaskReduxDispatch) => {
-    log.debug(`background.delRpcTarget: ${oldRpcUrl}`);
+export function editAndSetNetworkConfiguration({
+  networkConfigurationId,
+  rpcUrl,
+  chainId,
+  nickname,
+  rpcPrefs,
+  ticker = EtherDenomination.ETH,
+}: {
+  networkConfigurationId: string;
+  rpcUrl: string;
+  chainId: string;
+  nickname: string;
+  rpcPrefs: RPCDefinition['rpcPrefs'];
+  ticker: string;
+}): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return async (dispatch) => {
+    log.debug(
+      `background.removeNetworkConfiguration: ${networkConfigurationId}`,
+    );
     try {
-      submitRequestToBackground('delCustomRpc', [oldRpcUrl]);
+      await submitRequestToBackground('removeNetworkConfiguration', [
+        networkConfigurationId,
+      ]);
     } catch (error) {
       logErrorWithMessage(error);
       dispatch(displayWarning('Had a problem removing network!'));
       return;
     }
 
     try {
-      await submitRequestToBackground('updateAndSetCustomRpc', [
-        newRpcUrl,
-        chainId,
-        ticker ?? EtherDenomination.ETH,
-        nickname || newRpcUrl,
-        rpcPrefs,
+      await submitRequestToBackground('upsertNetworkConfiguration', [
+        {
+          rpcUrl,
+          chainId,
+          ticker,
+          nickname: nickname || rpcUrl,
+          rpcPrefs,
+        },
+        { setActive: true, source: ORIGIN_METAMASK },
```

Comment:
Hmm, this doesn't seem right. `ORIGIN_METAMASK` would be the referrer. For source, this appears to only be called from the network form, though I'm not sure whether we should hard-code that here (e.g. maybe we can pass it through instead, in case someone uses this in another context).

Reply:
Whoops! Another spot I missed! Good catch!

Reply:
fixed here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/9a9c2354f4692e4b377babfe97f09d602756fc68